### PR TITLE
feat: use bgp router annotations from node when creating cilium node

### PR DESCRIPTION
**Pull request by margau**
Please ensure your pull request adheres to the following guidelines:

- [x] For first time contributors, read [Submitting a pull request](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#submitting-a-pull-request)
- [x] All code is covered by unit and/or runtime tests where feasible.
- [x] All commits contain a well written commit description including a title,
      description and a `Fixes: #XXX` line if the commit addresses a particular
      GitHub issue.
- [ ] If your commit description contains a `Fixes: <commit-id>` tag, then
      please add the commit author[s] as reviewer[s] to this issue.
- [x] All commits are signed off. See the section [Developer’s Certificate of Origin](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#dev-coo)
- [x] Provide a title or release-note blurb suitable for the release notes.
- [ ] Are you a user of Cilium? Please add yourself to the [Users doc](https://github.com/cilium/cilium/blob/master/USERS.md)
- [ ] Thanks for contributing!

When using cluster-pool, a new CiliumNode Resource is created. This also needs the BGP Control Plane Annotations if present. This PR copies the annotation from the Node to the CiliumNode at creation time.

```release-note
Use BGP Control Plane annotations from Node Resource for creation of CiliumNode Resource
```

### CODE REVIEW

File: pkg/node/types/node.go [hunk 1]
```diff
@@ -129,6 +130,13 @@ func (n *Node) ToCiliumNode() *ciliumv2.CiliumNode {
 		annotations[annotation.WireguardPubKey] = n.WireguardPubKey
 	}
 
+	// if we use a cilium node instead of a node, we also need the BGP Control Plane annotations in the cilium node instead of the main node
+	for k, a := range n.Annotations {
+		if strings.HasPrefix(k, "cilium.io/bgp-virtual-router.") {
```
Comment (pippolo84) [inline pkg/node/types/node.go]: I suggest to use the `BGPVRouterAnnoPrefix` constant instead.
Due to a potential import cycle, to do so we should first move `BGPVRouterAnnoPrefix` from `pkg/bgpv1/agent/annotations.go` to `pkg/annotation/k8s.go` and then reference it here.

Even if moving the annotation outside of its "proper" location in the codebase is something I don't particularly like, I think it is better than "duplicating" the string.

**Review by pippolo84 — CHANGES_REQUESTED** [review]
Overall the PR looks good, thanks! I left a comment about the annotation prefix.

File: pkg/node/types/node.go (hunk 1 above)
Comment (margau) [reply pkg/node/types/node.go]: I've moved the definition

**Review by pippolo84 — APPROVED** [review]
LGTM, thanks! :rocket:

File: pkg/node/types/node_test.go [hunk 2]
```diff
@@ -197,6 +197,9 @@ func (s *NodeSuite) TestNode_ToCiliumNode(c *C) {
 		IPv6IngressIP:           net.ParseIP("c0de::2"),
 		NodeIdentity:            uint32(12345),
 		WireguardPubKey:         "6kiIGGPvMiadJ1brWTVfSGXheE3e3k5GjDTxfjMLYx8=",
+		Annotations: map[string]string{
+			"cilium.io/bgp-virtual-router.64512": "router-id=172.0.0.3",
```
Comment (pippolo84) [inline pkg/node/types/node_test.go]: Nit: it'd be better to update the test too with `annotation.BGPVRouterAnnoPrefix`.

File: pkg/node/types/node_test.go [hunk 3]
```diff
@@ -205,7 +208,8 @@ func (s *NodeSuite) TestNode_ToCiliumNode(c *C) {
 			Name:      "foo",
 			Namespace: "",
 			Annotations: map[string]string{
-				annotation.WireguardPubKey: "6kiIGGPvMiadJ1brWTVfSGXheE3e3k5GjDTxfjMLYx8=",
+				annotation.WireguardPubKey:           "6kiIGGPvMiadJ1brWTVfSGXheE3e3k5GjDTxfjMLYx8=",
+				"cilium.io/bgp-virtual-router.64512": "router-id=172.0.0.3",
```
Comment (pippolo84) [inline pkg/node/types/node_test.go]: Nit: same here.

**Review by ldelossa — APPROVED** [review]
Changes look good to me.

File: pkg/node/types/node_test.go (hunk 2 above)
Comment (margau) [reply pkg/node/types/node_test.go]: Using the prefix with a hardcoded test suffix now.

**Review by joamaki — APPROVED** [review]

**Comment by margau** [comment]
Hi, can someone with appropriate rights reapprove the pipelines? The commit length issue from previous checks should be fixed. Thanks!

**Comment by joestringer** [comment]
There's a few test failures here.

- ConformanceGKE hit https://github.com/cilium/cilium/issues/25267
- k8s-1.26-kernel-net-next hit https://github.com/cilium/cilium/issues/25119
- Runtime hit a failure in running the unit tests, not sure if related to this PR or not. I've filed a new issue here to initiate discussion: https://github.com/cilium/cilium/issues/25276

**Comment by joestringer** [comment]
/test-1.26-net-next

**Comment by joestringer** [comment]
The runtime tests have failed in the same way for a second time in a row, it seems likely that there is some cause related to the code changes here. Can you take a look?

I think it should be possible to reproduce with `sudo -E make tests-privileged TESTPKGS=./pkg/datapath/linux`.

**Comment by joestringer** [comment]
Ah, correction, it's a different package this time. Either way, can you see whether you observe these issues locally?

**Comment by margau** [comment]
Locally, this test seems to work:
```
OK: 36 passed
--- PASS: Test (238.17s)
=== RUN   FuzzNodeHandler
--- PASS: FuzzNodeHandler (0.00s)
PASS
	github.com/cilium/cilium/pkg/datapath/linux	coverage: 55.0% of statements
ok  	github.com/cilium/cilium/pkg/datapath/linux	238.197s	coverage: 55.0% of statements
```

**Comment by joestringer** [comment]
OK, thanks for taking a look. I've filed https://github.com/cilium/cilium/issues/25291 for that latest failure. Given that we haven't seen reports for that test failing otherwise, I'll re-run that test once more to gain further insights.

**Pull request merged**